src/js/player.js

```diff
@@ -1747,10 +1747,11 @@ class Player extends Component {
    */
   duration(seconds) {
     if (seconds === undefined) {
-      return this.cache_.duration || 0;
+      // return NaN if the duration is not known
+      return this.cache_.duration !== undefined ? this.cache_.duration : NaN;
     }
 
-    seconds = parseFloat(seconds) || 0;
+    seconds = parseFloat(seconds);
```

Comment:
[This conditional](https://github.com/videojs/video.js/blob/c9022795fdfcda166a79bef51dfeab29c1f7c191/src/js/control-bar/time-controls/duration-display.js#L74) will be `false` if `player.duration() = NaN`, so off the top of my head I can't think of a problem going from a unknown duration to a known duration. If we are transitioning from a known duration to an unknown duration, I believe the duration display (as currently written) would keep it's previous 'known' state until the new duration becomes known and the display updates. Assuming I'm right about that, would this be acceptable behavior?